### DISCUSSION TITLE
Private Git Dag Links Force you to Kill Worker Pods

### DISCUSSION THREAD

**Discussion by drcrook1** [apache/airflow#24497] [General]
### Apache Airflow version

2.3.2 (latest released)

### What happened

Deployed a helm upgrade of: apache-airflow/airflow with added private git repo dags.  This worked, however, I had to have the kubernetes admin kill the worker pods because they entered a non recoverable failure state.  Once they were killed, they booted back up perfectly happy.

This leads me to a suspicion that if the private git dag worker (excuse my lack of inner workings here of airflow) if it fails will cause the workers to go back into a failed state.

This is something which is highly un-desireable.  I do very much like being able to link multiple private git repos so all teams who are building dags can have their dags link up.

### What you think should happen instead

When adding a private repo, the workers should not enter a failed state.

### How to reproduce

helm upgrade --install airflow apache-airflow/airflow --namespace airflow --create-namespace `
    -f override-values.yaml

ovverride-values.yaml contents below:

dags:
  gitSync:
    enabled: true
    repo: ssh://git@github.com/youruserid/yourrepo.git
    branch: main
    subPath: ""
    sshKeySecret: airflow-ssh-secret
extraSecrets:
  airflow-ssh-secret:
    data: |
      gitSshKey: 'your key'

### Operating System

windows 11

### Versions of Apache Airflow Providers

default everything.

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

Deployed to Azure Kubernetes Service in East US.

### Anything else

Happens so far every time.  I'm working on the automated deployments, and it seems pretty consistent every time I blow out the deployment and do it again.

I suppose I could submit a PR.  I don't know much about this system but if somebody wants to buddy up with me, that would be pretty cool.  My day job would likely give me the bandwidth to do this.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Reply by potiuk** [reply]
I think youy have a problem with your deployment - it's not airflow nor helm issue, but likely the way you use git-sync configuration (if this is at all git-sync caused). Your diagnosis that private reapo is the root cause is completely unfounded, or at least nothing that you explained here leads to the conclusion that "Airflow has problems with private git links".

The way how to approach is you need to take a look at the logs of your Kubernetes pods and figure out why you experience what you experience. Kubernetes, Pods and containers each have logs that can be looked at to determine the root cause of your problem. 

By doing it, you will likely find out what caused the problem, or at least gather enough evidence so that you can provide us anough information that can help us to help with your problem.

Converted it into discussion and awaiting result of your investigtion to be able to help you with your diagnosis.
